setup/env.sh

```diff
@@ -475,14 +475,29 @@ if [[ $LLMDBENCH_CONTROL_CALLER == "run.sh" ]]; then
   fi
 fi
 
-required_vars=("LLMDBENCH_HF_TOKEN")
-for var in "${required_vars[@]}"; do
-  if [ -z "${!var:-}" ]; then
-    echo "❌ Environment variable '$var' is not set."
-    exit 1
+export HF_TOKEN=${HF_TOKEN:-$LLMDBENCH_HF_TOKEN}
+
+if ! echo ${LLMDBENCH_CONTROL_CALLER} | grep -iq "teardown"; then
+  if is_hf_model_gated "${LLMDBENCH_DEPLOY_MODEL_LIST}"; then
+    if [[ -z ${HF_TOKEN} ]]; then
+      announce "❌ Hugging Face Token is empty but attempted to use gated model \"${LLMDBENCH_DEPLOY_MODEL_LIST}\""
+      exit 1
+    fi
+
+    if user_has_hf_model_access "${LLMDBENCH_DEPLOY_MODEL_LIST}" "${HF_TOKEN}"; then
```

Comment:
`LLMDBENCH_DEPLOY_MODEL_LIST` could be a list of models, comma separated. Need for loop here?